### ISSUE TITLE
[New Item Variant] Sphere and franken-type piston and track platforms - creates platform topped with grating

### ISSUE THREAD

**Issue by Critfish** [BEEmod/BEE2-items#3170]
With the current way Track and Piston platforms are set up, you can use the Cube Type property to change the surface on the platform. I propose adding two new types - unportalable and portalable, each topped with a grate

-Standard - Default glass top
-Companion - Unportalable
-Reflective - Portalable
-Sphere - Unportalable, with grate, could be useful for Conversion Gel puzzles
-Franken- Portalable, with grate

**Comment by Flash4433** [comment]
So... a track/piston platform with a grate on top? So only gels and Portal shots can interact with the platform/metal itself?

**Comment by Critfish** [comment]
Yes. You can do this with regular grates and surfaces, and there are even actual reclined grate items in BEE2.4. It would be good for track/piston platforms to have this kind of functionality too. Conveniently, there are two unused options(Sphere and Franken), which would allow for unportalable and portalable platforms topped with grates.

**Comment by King0191** [comment]
Unfortunately, if i recall correctly (I may be wrong. Do not quote me on this) Franken sends out an error message? I think? That may only be in certain cases tho.

**Comment by Luke18033** [comment]
If it throws a compile error, that would probably just be because there's no variant for it (which this would fix). I'm pretty sure though that TeamSpen was planning on using these for an anti-gel surface item he was developing, not grating topped versions.

**Comment by King0191** [comment]
I meant just in general with the Franken option, that it broke things. I could totally be off tho, mighta been fixed? idk
